org_chart.js

```diff
@@ -1,73 +1,56 @@
-    project: "Compilers"
-  },
-  {
-    name: "Donald Knuth",
-    email: "donald.knuth@musky.com",
-    position: "Mathematician",
-    project: "Algorithm Analysis"
-  }
-]
-
-// assign each a manager
-let [alan, grace, donald] = new_hires
-
-alan.manager = elon
-grace.manager = elon
-donald.manager = elon
-
-// send to training
-if (!alan.skills) {
-  alan.skills = []
-}
-alan.skills.push("Mathematics")
-
-if (!grace.skills) {
-  grace.skills = []
+Hire.prototype.train = function(skill) {
+    this.skills.push(skill);
+    return this;
 }
-grace.skills.push("Cobol")
-
-if (!donald.skills) {
-  donald.skills = []
-}
-donald.skills.push("Computational Complexity")
-
-// whoops!  new person hired later on, do the same again for him
-let new_hire = {
-  name: "Tim Berners-Lee",
-  email: "tim.berners_lee@musky.com",
-  position: "Computer Science",
-  project: "Networks"
+Hire.prototype.addManager = function(employee_object) {
+    this.manager = employee_object;
 }
-let tim = new_hire
-tim.manager = elon
-
-if (!tim.skills) {
-  tim.skills = []
-}
-tim.skills.push("Protocols")
-
-console.log(`${alan.name} is a ${alan.position} working on ${alan.project}.`)
-console.log(`He can be reached at ${alan.email}`)
-console.log(`${grace.name} is a ${grace.position} working on ${grace.project}.`)
-console.log(`She can be reached at ${grace.email}`)
-console.log(`${donald.name} is a ${donald.position} working on ${donald.project}.`)
-console.log(`He can be reached at ${donald.email}`)
-console.log(`${tim.name} is a ${tim.position} working on ${tim.project}.`)
-console.log(`He can be reached at ${tim.email}`)
+////////////////////////////////////////////
+
+//Instance Specific/////////////////////////
+let employees = [],
+    candidates = [
+        ['Elon Musk', 'Boss', '', employees],
+        ['Alan Turing', 'Computer Scientist', 'Cryptography', employees],
+        ['Grace Hopper', 'Computer Scientist', 'Compilers', employees],
+        ['Donald Knuth', 'Mathematician', 'Algorithm Analiysis', employees],
+        ['Tim Berners-Lee', 'Computer Science', 'Networks', employees]
+    ];
+
+candidates.forEach(candidate => {
+    Reflect.construct(Hire, candidate)
+})
+
+let trainees = [
+        ['Alan Turing', 'Mathematics'],
+        ['Grace Hopper', 'Cobol'],
+        ['Donald Knuth', 'Computational Complexity'],
+        ['Tim Berners-Lee', 'Protocols']
+    ],
+    index;
+
+trainees.forEach(trainee => {
+    index = employees.findIndex(employee => employee.name === trainee[0]);
+    employees[index].train(trainee[1])
+})
+
+employees.forEach(employee => {
+    if (employee.position != 'Boss') {
+        employee.addManager(employees.find(employee => employee.position === 'Boss'))
+    }
+})
+employees.forEach(employee => {
+    if (employee.position != 'Boss') {
+        console.log(`${employee.name} is a ${employee.position} working on ${employee.project}`)
+        console.log(`and can be reached at ${employee.email}`)
```

Comment:
Notice that these two lines are exclusively about the employee.  This would hint that this would be another good candidate to be an instance method of the object.